ui/src/app/components/apps-panel/expanded-app.tsx

```diff
@@ -160,18 +166,24 @@ const CromwellButtonRow = (props: {
   );
 };
 
-const RStudioButtonRow = (props: {
+interface rstudioProps extends NavigationProps {
   userApp: UserAppEnvironment;
-  workspaceNamespace: string;
-}) => {
-  const { userApp, workspaceNamespace } = props;
-
+  workspace: WorkspaceData;
+}
+const RStudioButtonRow = fp.flow(
+  withNavigation,
```

Comment:
what about `useNavigation()` instead?

Reply:
using store for both navigation and workspace